Automattic/wp-calypso
# Site Migration: Upgrade plan step add dynamic faster response based on site FID

Based off of https://github.com/Automattic/wp-calypso/pull/90457



Related to https://github.com/Automattic/wp-calypso/issues/88971

## Proposed Changes

* Replace "Reduced error rate" (to be discussed) with "Faster response" item on the Site migration upgrade plan step. Google considers a  First Input Delay (FID) of 100 milliseconds or less a "Good" FID value. If the FID of the imported site is greater than 100 milliseconds, then we display the updated text with the speed percentage difference.
* Different from the proposed design, I also added how faster it is in WPCOM, following the same idea from https://github.com/Automattic/wp-calypso/pull/90457. To be discussed if it makes sense or if we should follow the design and don't show how faster it is in WPCOM.
* It also adds a skeleton loader, so we don't show the fallback if we will present other information.

With FID lower than 100ms:

<img width="462" alt="Screenshot 2024-05-15 at 11 58 57" src="https://github.com/Automattic/wp-calypso/assets/876340/a4f1bd11-bf3d-4d14-b307-f183b7f7baf3">

With FID greater than 100ms:

<img width="371" alt="Screenshot 2024-05-15 at 16 24 27" src="https://github.com/Automattic/wp-calypso/assets/876340/75f63e7e-d5d0-4cc2-a8a3-a04f49dbc0af">

## Known issues

* https://github.com/Automattic/wp-calypso/issues/90715

## Testing Instructions



* Checkout branch or use the live link.
* Navigate to both links: `/setup/site-setup/importerWordpress?siteSlug={FREE_SITE_SLUG}&from=https://www.mazdausa.com&option=everything` and `/setup/site-migration/site-migration-upgrade-plan?from=https://www.mazdausa.com&siteSlug={FREE_SITE_SLUG}`
* Verify that nothing changes compared with `update/upgrade-plan-hosting-dynamic-data-speed` because the sample site has a good FID.
* Manually update the line `siteMetricData?.basic?.fid` to `150` (a not good FID).
* Verify that "Reduced error rate" was replaced with with "Faster response".
* Verify faster response difference formula is returning correct value.

## Pre-merge Checklist
